File: 1927-sum-game/1927-sum-game.c
#include <stdbool.h>
#include <string.h>

bool sumGame(char* num) {
    int n = strlen(num);
    int half = n / 2;

    int leftSum = 0;
    int rightSum = 0;

    int qLeft = 0;
    int qRight = 0;

    for (int i = 0; i < half; i++) {
        char c = num[i];

        if (c == '?') {
            qLeft++;
        } else {
            leftSum += c - '0';
        }
    }

    for (int i = half; i < n; i++) {
        char c = num[i];

        if (c == '?') {
            qRight++;
        } else {
            rightSum += c - '0';
        }
    }

    int diff = leftSum - rightSum;

    return 2 * diff != 9 * (qRight - qLeft);
}

File: 1927-sum-game/1927-sum-game.js
/**
 * @param {string} num
 * @return {boolean}
 */
var sumGame = function(num) {
    const n = num.length;
    const half = Math.floor(n / 2);

    let leftSum = 0;
    let rightSum = 0;

    let qLeft = 0;
    let qRight = 0;

    for (let i = 0; i < half; i++) {
        const c = num[i];

        if (c === '?') {
            qLeft++;
        } else {
            leftSum += c.charCodeAt(0) - 48; // '0' is 48
        }
    }

    for (let i = half; i < n; i++) {
        const c = num[i];

        if (c === '?') {
            qRight++;
        } else {
            rightSum += c.charCodeAt(0) - 48;
        }
    }

    const diff = leftSum - rightSum;

    return 2 * diff !== 9 * (qRight - qLeft);
};
